src/collective/taxonomy/utility.py

```diff
@@ -291,9 +291,7 @@ def translate(self, msgid, mapping=None, context=None,
 
         if target_language is None or \
                 target_language not in self.inverted_data:
-            target_language = str(self.getCurrentLanguage(
-                getattr(context, 'REQUEST')
-            ))
+            target_language = str(api.portal.get_current_language())
```

Comment:
`self.getCurrentLanguage` leads to `collective.taxonomy.interfaces.get_lang_code` and this gets the language by plone.api ... is this really needed here?

Reply:
there was no context in some cases